Sources/FirebaseFirestore/Transaction+Swift.swift

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+@_exported
+import firebase
+
+import CxxShim
+import Foundation
+
+public typealias Transaction = swift_firebase.swift_cxx_shims.firebase.firestore.TransactionWeakReference
+
+extension Transaction {
+  public mutating func setData(_ data: [String : Any], forDocument document: DocumentReference) -> Transaction {
+    setData(data, forDocument: document, merge: false)
+  }
+
+  public mutating func setData(_ data: [String : Any], forDocument document: DocumentReference, merge: Bool) -> Transaction {
+    assert(is_valid())
```

Comment:
I think in some other PRs we were using `fatalError`s when something wasn't valid. Do we want this to crash in release or are we okay with some different behavior? I'd argue to crash if we're in a state where we shouldn't be, but I could also just be misunderstanding the Transaction APIs

Reply:
Yeah, I changed this to be a `fatalError` in the follow-up PR.